archunit/src/test/java/com/tngtech/archunit/library/ArchitecturesTest.java

```diff
@@ -134,24 +134,33 @@ public void layered_architecture_defining_constraint_on_non_existing_origin_is_r
     }
 
     @Test
-    public void layered_architecture_defining_empty_layers_is_rejected() {
+    @DataProvider(value = {"true", "false", "null"})
+    public void layered_architecture_allows_or_rejects_empty_layers(Boolean allowEmptyLayers) {
         LayeredArchitecture architecture = layeredArchitecture()
                 .layer("Some").definedBy(absolute("should.not.be.found.."))
                 .layer("Other").definedBy(absolute("also.not.found"))
                 .layer("Okay").definedBy("..testclasses..");
+        if (allowEmptyLayers != null) {
```

Comment:
Same as below. For me it is also hard to immediately grasp at the first look, that those two conditional blocks relate to each other and are only set in conjunction...

Reply:
There's no 1:1 correspondence, but an explicit test is probably still better than this implicit construction.